src/notifications/util.js

```diff
@@ -1,23 +1,19 @@
-async function tryCallbackOrPromise(promise, callback = null) {
+function tryCallbackOrReturnResult(data, callback = null) {
```

Comment:
This function is called tryCallbackOrReturnResult so I think data which represent a promise (where We dont know if is resolved or not) but we can call like `promise` or `result`, what do you think?